### ISSUE TITLE
Sanic version incompatible with Rasa

### ISSUE THREAD

**Issue by tayfun** [RasaHQ/rasa#10632]
### Rasa Open Source version

3.0.x

### Python version

3.8

### What operating system are you using?

OSX

### What happened?

We have a [ticket showing new Sanic versions which are compatible with `pyproject.toml` breaking rasa-sdk](https://github.com/RasaHQ/rasa-sdk/issues/647). In addition to that, I find we have other incompatibilities with rasa. Currently we have `sanic = "^21.6.0"` in pyproject.toml file. In pypi we don't use lock files, poetry.lock file is for local dev only. If I update requirements and run shell, I get import error on `CompositionView`.

```
poetry update
```

Results in Sanic being upgraded to `21.12.0` from `21.9.3`. When I run shell, I get following exception (app name exception should come later):

```
$ rasa shell --help
Traceback (most recent call last):
  File "/Users/tayfun/.virtualenvs/rasa3.0/bin/rasa", line 2, in <module>
    from rasa.__main__ import main
  File "/Users/tayfun/w/rasa/rasa/__main__.py", line 14, in <module>
    from rasa.cli import (
  File "/Users/tayfun/w/rasa/rasa/cli/export.py", line 9, in <module>
    import rasa.core.utils
  File "/Users/tayfun/w/rasa/rasa/core/utils.py", line 25, in <module>
    from sanic.views import CompositionView
ImportError: cannot import name 'CompositionView' from 'sanic.views' (/Users/tayfun/.virtualenvs/rasa3.0/lib/python3.8/site-packages/sanic/views.py)
```

Once https://github.com/RasaHQ/rasa/pull/10590/files is merged, check shell run again and see if Sanic name should be updated to be without any dots. After fixing Sanic app name, we should also check any other incompatibilities. If too many, we can also pin down Sanic in pyproject, but this has to be consistent with rasa-sdk.

**Definition of done**
- [ ] Pin to a minor version in Rasa and Rasa SDK
- [ ] Close https://github.com/RasaHQ/rasa-sdk/issues/647

**Comment by koaning** [comment]
This is potentially a duplicate of https://github.com/RasaHQ/rasa/issues/10585

**Comment by koaning** [comment]
I just saw this bug re-appear on Rasa Open Source 3.0.5.
